### PR TITLE
Adding access key signer that uses wallet for auth

### PR DESCRIPTION
**Pull request by evgenykuzyakov**
It has the same API as WalletAccount, so it should be easy to upgrade.
It doesn't use `network_id` by default (uses default from key storage), so we may update API if needed.
It needs wallet changes to provide access point `login_v2` that would add a new access_key to the account and return given `public_key` instead of `auth_token`.

### CODE REVIEW

File: nearlib/wallet-access-key.js [hunk 1]
```diff
@@ -0,0 +1,128 @@
+/**
+ * Wallet based account and signer that uses external wallet through the iframe to signs transactions.
```
Comment (janedegtiareva) [inline nearlib/wallet-access-key.js]: nit s/signs/sign

Comment (janedegtiareva) [inline nearlib/wallet-access-key.js]: Is the iframe comment still correct?

Comment (evgenykuzyakov) [reply nearlib/wallet-access-key.js]: Oh. Didn't update that top level comment.

File: nearlib/API.md [hunk 2]
```diff
@@ -108,7 +108,7 @@ Generate a new keypair from a random seed
 #### Examples
 
 ```javascript
-const keyWithRandomSeed = await KeyPair.fromRandomSeed();
```
Comment (vgrichina) [inline nearlib/API.md]: nit: it's generally better to keep APIs with potentially heavy compute as async (cause then you can replace implementation on something that doesn't block threads)

File: nearlib/wallet-access-key.js [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+        return this._authData.accountId || '';
+    }
+
+    /**
+     * Redirects current page to the wallet authentication page.
+     * @param {string} contract_id contract ID of the application
+     * @param {string} title name of the application
+     * @param {string} success_url url to redirect on success
+     * @param {string} failure_url url to redirect on failure
+     * @example
```
Comment (vgrichina) [inline nearlib/wallet-access-key.js]: let's not provide examples / comments which don't add extra information over method signature

File: nearlib/wallet-access-key.js [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+     *   walletAccount.requestSignIn(
+     *     myContractId,
+     *     title,
+     *     onSuccessHref,
+     *     onFailureHref);
+     */
+    requestSignIn(contract_id, title, success_url, failure_url) {
```
Comment (vgrichina) [inline nearlib/wallet-access-key.js]: snake_case -> camelCase

File: nearlib/API.md (hunk 2 above)
Comment (evgenykuzyakov) [reply nearlib/API.md]: Oh, I see. The reason for this change was keep API the same as WalletAccount. But I can just wrap logic with `.then(...)`

**Pull request merged**